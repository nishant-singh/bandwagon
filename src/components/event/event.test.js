import React from 'react';
import { shallow, mount } from 'enzyme';

import Event from './event';
import dateConverter from "../../utils/convertToDate";

const event = {
    venue:{
        name: "My testing audition",
        city: "Berlin",
        country: "Germany"
    },
    datetime : "2018-10-10 20:11:00"
}
describe('Event', () => {
    const component = mount( < Event data={event}/ > );
    test('should render correctly in "debug" mode', () => {
        const component = shallow( < Event data={event}/ > );
        expect(component).toMatchSnapshot();
    });
    test('render complete event', () => {
        const event = component.find('.event');
        expect(event.length).toBe(1);
    });
    test('renders event venue', () => {
        const venue = component.find('.venue');
        expect(venue.length).toBe(1);
        expect(venue.text()).toBe(event.venue.name);
    });
    test('renders event location', () => {
        const location = component.find('.venue_location');
        expect(location.length).toBe(1);
        expect(location.text()).toBe(`${event.venue.city}, ${event.venue.country}`);
    });
    test('renders event date', () => {
        const date = component.find('.venue_date');
        expect(date.text()).toBe(dateConverter(event.datetime));
    });
});


