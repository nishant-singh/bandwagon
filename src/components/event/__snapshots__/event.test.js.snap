// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventDetails
    data={
        Object {
            "datetime": "2018-10-10 20:11:00",
            "venue": Object {
              "city": "Berlin",
              "country": "Germany",
              "name": "My testing audition",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="row"
>
          <div
                    className="venue"
          >
                    My testing audition
          </div>
          <div
                    className="venue_location"
          >
                    <img
                              alt="location"
                              className="icon"
                              src="location_icon.png"
                    />
                    Berlin
                    , 
                    Germany
          </div>
</section>,
        <section
          className="row venue_date"
>
          Wed Oct 10 2018, 8:11 PM
</section>,
      ],
      "className": "event",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="venue"
>
              My testing audition
</div>,
            <div
              className="venue_location"
>
              <img
                            alt="location"
                            className="icon"
                            src="location_icon.png"
              />
              Berlin
              , 
              Germany
</div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "My testing audition",
              "className": "venue",
            },
            "ref": null,
            "rendered": "My testing audition",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="location"
                  className="icon"
                  src="location_icon.png"
/>,
                "Berlin",
                ", ",
                "Germany",
              ],
              "className": "venue_location",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "location",
                  "className": "icon",
                  "src": "location_icon.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "Berlin",
              ", ",
              "Germany",
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Wed Oct 10 2018, 8:11 PM",
          "className": "row venue_date",
        },
        "ref": null,
        "rendered": "Wed Oct 10 2018, 8:11 PM",
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="row"
>
            <div
                        className="venue"
            >
                        My testing audition
            </div>
            <div
                        className="venue_location"
            >
                        <img
                                    alt="location"
                                    className="icon"
                                    src="location_icon.png"
                        />
                        Berlin
                        , 
                        Germany
            </div>
</section>,
          <section
            className="row venue_date"
>
            Wed Oct 10 2018, 8:11 PM
</section>,
        ],
        "className": "event",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="venue"
>
                My testing audition
</div>,
              <div
                className="venue_location"
>
                <img
                                alt="location"
                                className="icon"
                                src="location_icon.png"
                />
                Berlin
                , 
                Germany
</div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "My testing audition",
                "className": "venue",
              },
              "ref": null,
              "rendered": "My testing audition",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="location"
                    className="icon"
                    src="location_icon.png"
/>,
                  "Berlin",
                  ", ",
                  "Germany",
                ],
                "className": "venue_location",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "location",
                    "className": "icon",
                    "src": "location_icon.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "Berlin",
                ", ",
                "Germany",
              ],
              "type": "div",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Wed Oct 10 2018, 8:11 PM",
            "className": "row venue_date",
          },
          "ref": null,
          "rendered": "Wed Oct 10 2018, 8:11 PM",
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
